scripts/SREP/get-sosreport/script.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+set -e
+set -o nounset
+set -o pipefail
+
+## validate input
+if [[ -z "$NODE" ]]
+then
+    echo "Variable node cannot be blank"
+    exit 1
+fi
+
+# Import sftp_upload library
+source /managed-scripts/lib/sftp_upload/lib.sh
+
+# Define expected values
+DUMP_DIR="${PWD}"
+DATE="$(date -u +"%Y%m%dT%H%M")"
+SOSREPORT_FILENAME="sosreport-${NODE}.tar.xz"
+SOSREPORT_FILEPATH="${DUMP_DIR}/sosreport-${NODE}.tar.xz"
+
+
+
+check_node(){
+    echo "Checking if \"${NODE}\" is an existing node..."
+    
+    if (oc get nodes -l node-role.kubernetes.io/worker= -oname | grep "${NODE}") &> /dev/null; then
+       echo "[OK] \"${NODE}\" is a node."
+    else
+        echo "[Error] \"${NODE}\" is not a node. Exiting script"
+        exit 1
+    fi
+}
+
+generate_sosreport() {
+# 1st Debug session - Generate the sosreport and keep it in the host
+
+    echo " ==== Generating SOSREPORT ===="
+    oc -n default debug node/"${NODE}" -- sh -c "chroot /host toolbox sos report -k crio.all=on -k crio.logs=on --batch"
+
+    return 0
+}
+
+validate_file() {
+    echo "==== Check if the sosreport is created inside the node ===="
+    oc -n default debug node/"${NODE}" -- bash -c "ls -l /host/var/tmp/*.tar.xz" | tee;
+
+    return 0
```

Comment:
You might not need to explicitly include `return 0` since the function can return the exit status of the command executed, I am thinking if it's better you can improve error handling for this function as well as other functions in the script, something like:  
```
if ! oc -n default debug node/"${NODE}" -- bash -c "ls -l /host/var/tmp/*.tar.xz" | tee;  then
        echo "Error: sosreport file not found on node"
        exit 1
fi
```